profiling/src/lib.rs

```diff
@@ -634,11 +635,16 @@ extern "C" fn rinit(_type: c_int, _module_number: c_int) -> ZendResult {
                 profiler.add_interrupt(interrupt);
             }
         });
+    } else {
+        TAGS.with(|cell| cell.replace(Arc::default()));
```

Comment:
Did you need this for any particular reason? It makes sense theoretically, just wondering about specifics.

Reply:
I think we might not need it after all. I moved it from `RSHUTDOWN` to here for just the one reason of having the `TAGS` in a defined state when used. If we do not cleanup, `TAGS` leak from the last request that had profiling enabled and might be wrong. Now the thing is: we are not using the `TAGS` when profiling is disabled.
I am totally fine with removing this entirely, but I'd like to document this somewhere (at best in Rust itself, otherwise a comment on the `TAGS` variable should also work).
We can add those kind of information/documentation also in another PR.

Reply:
I think it's consistent. Previously we cleared tags in prshutdown and conditionally re-initialized them in rinit. Now we leave them alone in prshutdown, and either re-initialize them or reset them in rinit. These are roughly equivalent, except for the case that if you go from `prshutdown` to `mshutdown`, they'll still be valid, which was your motiviation. Seeems good to me.